astropy/modeling/physical_models.py

```diff
@@ -0,0 +1,187 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Models that have physical origins.
+"""
+
+import warnings
+from collections import OrderedDict
+
+import numpy as np
+
+from .core import Fittable1DModel
+from .parameters import Parameter
+from astropy import constants as const
+from astropy import units as u
+from astropy.utils.exceptions import AstropyUserWarning
+
+__all__ = ["BlackBody"]
+
+
+class BlackBody(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : :class:`~astropy.units.Quantity`
+        Blackbody temperature.
+
+    scale : :class:`~astropy.units.Quantity`
+        Scale factor
+
+    Notes
+    -----
+
+    Model formula:
+
+        .. math:: B_{\\nu}(T) = A \\frac{2 h \\nu^{3} / c^{2}}{exp(h \\nu / k T) - 1}
+
+    Examples
+    --------
+    >>> from astropy.modeling import models
+    >>> from astropy import units as u
+    >>> bb = models.BlackBody(temperature=5000*u.K)
+    >>> bb(6000 * u.AA)  # doctest: +FLOAT_CMP
+    <Quantity 1.53254685e-05 erg / (cm2 Hz s sr)>
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        from astropy.modeling.models import BlackBody
+        from astropy import units as u
+        from astropy.visualization import quantity_support
+
+        bb = BlackBody(temperature=5778*u.K)
+        wav = np.arange(1000, 110000) * u.AA
+        flux = bb(wav)
+
+        with quantity_support():
+            plt.figure()
+            plt.semilogx(wav, flux)
+            plt.axvline(bb.nu_max.to(u.AA, equivalencies=u.spectral()).value, ls='--')
+            plt.show()
+    """
+
+    # We parametrize this model with a temperature and a scale.
+    temperature = Parameter(default=5000.0, min=0, unit=u.K)
+    scale = Parameter(default=1.0, min=0)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be frequencies in Hz.
+    _input_units_allow_dimensionless = True
```

Comment:
Is there supposed to be a leading underscore? See https://docs.astropy.org/en/latest/modeling/add-units.html#input-units-allow-dimensionless

Reply:
Yes, the one you are pointing to is a property. There are still loose ends with units that we won't have time to fix for 4.0